pam-u2f.c

```diff
@@ -87,7 +87,7 @@ static void parse_cfg(int flags, int argc, const char **argv, cfg_t *cfg) {
         cfg->debug_file = (FILE *) -1;
       } else {
         fd = open(filename,
-                  O_WRONLY | O_APPEND | O_CLOEXEC | O_NOFOLLOW | O_NOCTTY);
+                  O_CREAT | O_WRONLY | O_APPEND | O_CLOEXEC | O_NOFOLLOW | O_NOCTTY, S_IRUSR | S_IRGRP);
```

Comment:
Duh. Ok, I read that part a couple of times but somehow seem to have missed this info. I guess it would be helpful to make this stick out even more. TBH, I already wondered why the O_CREAT flag has been missing in the first place. So, this is why. OTOH, I reflected briefly on the security implications on this and decided for myself I'd rather have debugging working. If debugging isn't enabled this is of no concern anyway. So maybe it make sense to just create the file if debug_file is set? If nothing else, common sense is telling that in production debugging has to be off, anyway? How do you think about it?

Reply:
I think there are people out there who turned on debugging, fixed the problem and then left debugging turned on to catch more potential problems and then promptly forgot about it because then it just worked. Then a reboot cleared /tmp and since then nothing got logged.

Changing this behaviour now could have the effect that existing installations would get a new pam-u2f package and the debug file would be created and then grow forever. Guess who they would blame? Also, this file can potentially contain sensitive information. Requiring that admins create this file themselves makes it clear who is responsible for cleaning up.

Reply:
Yes, I get your point.

I think instead of working around such an issue is no solution. In such a case I'd rather force-enable debug output on stderr like

DEBUG MODE ENABLED!

That way it's rather hard to miss it.

In the end, this issue wasn't my main incentive for the PR, it rather was the individually configurable user file issue.